lib/matplotlib/style/core.py

```diff
@@ -66,6 +67,12 @@ def use(style):
     else:
         styles = style
 
+    tmp_styles = []
+    for style in styles:
+        style = get_parents_styles(style)
+        tmp_styles.extend(style)
+    styles = tmp_styles[:]
```

Comment:
I think this should be refactored so this diff is a one-liner; e.g. `styles = insert_parent_styles(styles)` ("insert" sort of implies an in-place operation, so a different verb might be appropriate)